maestro-client/src/main/java/maestro/android/AndroidAppFiles.kt

```diff
@@ -37,6 +37,15 @@ object AndroidAppFiles {
         }
     }
 
+    fun getApkFile(dadb: Dadb, appId: String): File {
+        val apkPath = dadb.shell("pm list packages -f | grep $appId")
+            .output.substringAfterLast("package:").substringBefore("=$appId")
+        apkPath.substringBefore("=$appId")
+        val dst = File.createTempFile("tmp", ".apk")
+        dadb.pull(dst, apkPath)
```

Comment:
Just to confirm, does this work on non-root devices and for non-debuggable apps as well?

Reply:
> dadb.pull(dst, apkPath)

This requires a debuggable app.

Reply:
If so, I am afraid this is a no-go. It needs to work with any kind of app

Reply:
I checked it on few apps from apk mirror and it was working 🤔

Reply:
I have tried this as well with non-debuggable apps and I think I remember it working

Reply:
I stand corrected, seems to be a case only when trying to pull the app cache (probably with `run-as`)

Reply:
Tested this out on few other apps:

1. We are going to fallback on monkey command in case we have exception
2. Also fixed logic to detect which activity app belongs to.